test/server/cards/01_SOR/events/ForcedSurrender.spec.ts

```diff
@@ -0,0 +1,147 @@
+                expect(context.player2.handSize).toBe(2);
+
+                reset();
+
+                // Scenario 3: Damage dealt with overwhelm
+                context.player2.clickCard(context.mazKanata);
+                context.player2.clickCard(context.p1Base);
+
+                context.player1.clickCard(context.wampa);
+                context.player1.clickCard(context.mazKanata);
+
+                context.player2.passAction();
+
+                context.player1.clickCard(context.forcedSurrender);
+                expect(context.player2).toBeActivePlayer();
+                expect(context.player1.handSize).toBe(4);
+                expect(context.player2.handSize).toBe(0);
+
+                reset();
+
+                // Scenario 4: Damage dealt with event
+                context.player2.clickCard(context.mazKanata);
+                context.player2.clickCard(context.p1Base);
+
+                context.player1.moveCard(context.daringRaid, 'hand');
+                context.player1.clickCard(context.daringRaid);
+                context.player1.clickCard(context.p2Base);
+
+                context.player2.passAction();
+
+                context.player1.clickCard(context.forcedSurrender);
+                expect(context.player2).toBeActivePlayer();
+                expect(context.player1.handSize).toBe(4);
+                expect(context.player2.handSize).toBe(0);
+
+                reset();
+
+                // Scenario 5: Damage dealt with ability and to self
+                context.player2.clickCard(context.mazKanata);
+                context.player2.clickCard(context.p1Base);
+
+                context.player1.clickCard(context.sabineWren);
+                context.player1.clickPrompt('Deal 1 damage to each base');
+
+                context.player2.passAction();
+
+                context.player1.clickCard(context.forcedSurrender);
+                expect(context.player2).toBeActivePlayer();
+                expect(context.player1.handSize).toBe(4);
+                expect(context.player2.handSize).toBe(0);
+
+                reset();
+
+                // Scenario 6: Damage dealt with multiple sources
+                context.player2.clickCard(context.mazKanata);
+                context.player2.clickCard(context.p1Base);
+
+                context.player1.clickCard(context.sabineWren);
+                context.player1.clickPrompt('Deal 1 damage to each base');
+
+                context.player2.passAction();
+
+                context.player1.clickCard(context.wampa);
+                context.player1.clickCard(context.mazKanata);
+
+                context.player2.passAction();
+
+                context.player1.moveCard(context.daringRaid, 'hand');
+                context.player1.clickCard(context.daringRaid);
+                context.player1.clickCard(context.p2Base);
+
+                context.player2.passAction();
+
+                const p2BaseDamageBeforeForceSurrender = context.p2Base.damage;
```

Comment:
Sorry for the late comment, just thought of one other case it would be great to have as more of an integration test - can you add one more case where both players try to draw from an empty deck and take damage? Then we confirm that forced surrender does nothing. The damage attribution in that situation is a special case so it would be great to have some coverage for it. Probably easiest to set up as its own case with its own setup.

Reply:
How should the damage attribution work in that case? I tried having Yoda triggering the draw and the damage was attributed to Yoda's ability, so counting for Forced Surrender.

Reply:
Replied in chat